popcorn/compute.py

```diff
@@ -137,6 +153,14 @@ def func(a, i, j):
                 if not args.use_bias:
                     c = c - (1-c)/(N-2)
                 return (v*c).sum(1), (v1j*(c)).sum(0)[0:-1]
+        elif args.h2weight:
+            v1m = np.mean(h2weight)
+            def func(a, i, j):
+                h2w1 = h2weight[i:j]
+                v = h2w1/v1m
+                v1j = h2w1[-1]/v1m
+                c = a**2
```

Comment:
Can you add conditional bias removal here as in the other functions?

if not args.use_bias:
    c = c - (1-c)/(N-2)

I think this will still be necessary to reduce inflation with wide window settings.